From 97141cd96a0b9cbe43d355d400e358309bafdd25 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:10 -0400
Subject: [fs] nfsd: fix memory leak in nfsd4_encode_fattr()

Message-id: <1403560314-32525-69-git-send-email-bfields@redhat.com>
Patchwork-id: 84570
O-Subject: [Patch RHEL7 068/172] nfsd4: fix memory leak in nfsd4_encode_fattr()
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: "Yan, Zheng" <zheng.z.yan@intel.com>

fh_put() does not free the temporary file handle.

Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 18df11d0eacf67bbcd8dda755b568bbbd7264735
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 94b7f19..18881f3 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -2500,8 +2500,10 @@ out:
   security_release_secctx(context, contextlen);
 #endif /* CONFIG_NFSD_V4_SECURITY_LABEL */
  kfree(acl);
- if (tempfh)
+ if (tempfh) {
   fh_put(tempfh);
+  kfree(tempfh);
+ }
  return status;
 out_nfserr:
  status = nfserrno(err);
-- 
1.7.1